src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs

```diff
@@ -0,0 +1,535 @@
+            packArgs.OutputDirectory = PackageOutputPath;
+            packArgs.Serviceable = Serviceable;
+            packArgs.Suffix = VersionSuffix;
+            packArgs.Tool = IsTool;
+            packArgs.Symbols = IncludeSymbols;
+            packArgs.NoPackageAnalysis = NoPackageAnalysis;
+            packArgs.PackTargetArgs = new MSBuildPackTargetArgs()
+            {
+                TargetPaths = TargetPaths,
+                Configuration = Configuration,
+                AssemblyName = AssemblyName
+            };
+            packArgs.PackTargetArgs.TargetFrameworks = ParseFrameworks();
+            if (MinClientVersion != null)
+            {
+                Version version;
+                if (!Version.TryParse(MinClientVersion, out version))
+                {
+                    throw new ArgumentException("MinClientVersion is invalid");
+                }
+                packArgs.MinClientVersion = version;
+            }
+
+            InitCurrentDirectoryAndFileName(packArgs);
+            InitNuspecOutputPath(packArgs);
+
+            if (IncludeSource)
+            {
+                packArgs.PackTargetArgs.SourceFiles = GetSourceFiles(packArgs.CurrentDirectory);
+                packArgs.Symbols = IncludeSource;
+            }
+
+            PackCommandRunner.SetupCurrentDirectory(packArgs);
+
+            return packArgs;
+        }
+
+        private ISet<NuGetFramework> ParseFrameworks()
+        {
+            var nugetFrameworks = new HashSet<NuGetFramework>();
+            if (TargetFrameworks != null)
+            {
+                nugetFrameworks = new HashSet<NuGetFramework>(TargetFrameworks.Select(t => NuGetFramework.Parse(t)));
+            }
+            return nugetFrameworks;
+        } 
+
+        private PackageBuilder GetPackageBuilder(PackArgs packArgs)
+        {
+            PackageBuilder builder = new PackageBuilder();
+            builder.Id = PackageId;
+            if (PackageVersion != null)
+            {
+                NuGetVersion version;
+                if (!NuGetVersion.TryParse(PackageVersion, out version))
+                {
+                    throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, "PackageVersion string specified '{0}' is invalid.", PackageVersion));
+                }
+                builder.Version = version;
+            }
+            else
+            {
+                builder.Version = new NuGetVersion("1.0.0");
+            }
+
+            if (Authors != null)
+            {
+                builder.Authors.AddRange(Authors);
+            }
+
+            if (Tags != null)
+            {
+                builder.Tags.AddRange(Tags);
+            }
+
+            builder.Description = Description;
+            builder.Copyright = Copyright;
+            builder.ReleaseNotes = ReleaseNotes;
+            Uri tempUri;
+            if (Uri.TryCreate(LicenseUrl, UriKind.Absolute, out tempUri))
```

Comment:
A NuGetFramework parsing failure [here](https://github.com/NuGet/NuGet.Client/pull/899/files#diff-89dbfecd7235a4059d7d396dbe49c9aeR116) or [here](https://github.com/NuGet/NuGet.Client/pull/899/files#diff-89dbfecd7235a4059d7d396dbe49c9aeR128) would fail `Execute()` but any URI parsing failure here or below for non-empty, non-null values won't?